internal/jimm/role/role.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2024 Canonical.
+package role
+
+import (
+	"context"
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	jimmnames "github.com/canonical/jimm/v3/pkg/names"
+)
+
+// RoleStore defines a storage interface to persist roles.
+type RoleStore interface {
```

Comment:
if we are not mocking this interface in our units test, why do we need it exactly?

Reply:
It limits the methods exposed to the RoleManager